### PR TITLE
Drop experimental support for eip-615 and evm15.

### PR DESCRIPTION
**Pull request by ekpyron**
This, intentionally, doesn't try anything clever, but just drops it. In case things can be simplified on top of this, I'd suggest to do that separately.

### CODE REVIEW

**Comment by ekpyron** [comment]
Since this was a command line option, should it be a changelog entry? Or has it always be experimental, so no entry is fine? I don't think we had documentation for it and introducing it had no changelog entry either, if I see it correctly...

**Comment by chriseth** [comment]
I think it's good to have a changelog entry.

**Review by chriseth — DISMISSED** [review]

**Comment by ekpyron** [comment]
Added a changelog entry.

**Review by axic — DISMISSED** [review]
Looks good, you may want to squash some of the commits.

File: Changelog.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ Language Features:
 
 
 Compiler Features:
+ * Drop obsolete experimental support for EIP-615, resp. for the ``--machine evm15`` in the Command Line Interface.
```
Comment (chriseth) [inline Changelog.md]: obsolete? Is it really EIP-615 or just "EVM15"?

Comment (ekpyron) [reply Changelog.md]: I can drop "obsolete" - but the PR drops the EIP-615 instructions and their use as the evm15 machine, so not sure what you mean by "Is it really EIP-615 or just "EVM15"?"?

Comment (ekpyron) [reply Changelog.md]: I now changed it to just say, it drops ``--machine evm15``.

**Comment by ekpyron** [comment]
Squashed and reformulated changelog entry slightly.

**Review by chriseth — DISMISSED** [review]

**Comment by chriseth** [comment]
Ok, it probably was 615, but maybe also an outdated version of it.

**Comment by ekpyron** [comment]
Sorry, I just pushed again to change the changelog entry again (making it properly refer to the command line interface - that's the user-visible change this does anyways).

**Comment by ekpyron** [comment]
Rebased to solve changelog conflict.

**Review by chriseth — APPROVED** [review]

**Pull request merged**